Minimize some editor panels to prevent GUI overflow

arkology opened the pull request:
Alternative to https://github.com/godotengine/godot/pull/101828. 
Same as linked PR, may fix https://github.com/godotengine/godot/issues/26835 and fix https://github.com/godotengine/godot/issues/96479.

Aslo includes https://github.com/godotengine/godot/pull/102024 for now

I looked at (almost) every panel which could affect horizontal or vertical overflow when window size is minimal and did some tweaks of minimal size or another changes to minimize panel size (for example, use `FlowContainer` instead of `BoxContainer`, enable `ScrollContainer` scrolling).

For now only missing editor panel is `AnimationTree` (WIP). But I may overlooked another panels.

Also includes some concomitant changes like enabling `Label` word wrapping and fixing C++ C26819 warning (this could be reverted if not wanted).

May include some improper changes, so any suggestions are very appreciated. Also, it may look different than it does on my end, and therefore not fix the overflow, so it needs testing.

It could be split into several PRs if needed and changes are accepted.

|      | Before | After |
| --- | ---| --- |
| `AnimationPlayer` tab (vertical overflow) | ![image](https://github.com/user-attachments/assets/c2e468b4-d8f9-4916-932e-7110a305ca59) | ![image](https://github.com/user-attachments/assets/2002137f-77ae-4e49-a201-3a532e4dea28) |
| `Audio` tab (vertical overflow) | ![image](https://github.com/user-attachments/assets/c5a5c1c3-58e5-460d-8225-42797e57d863) | ![image](https://github.com/user-attachments/assets/61ee1d51-78b6-4809-802b-d72c3dc122fc) | 
| `AnimationTree` tab (vertical overflow, WIP) | ![image](https://github.com/user-attachments/assets/f02d08a7-a249-4e01-ae11-e3ad6449d002) | ![image](https://github.com/user-attachments/assets/8b91ced1-fef3-4960-86ec-a7bd0d8ea478) |
| Animation, `BlendSpace1D` (vertical overflow) | ![image](https://github.com/user-attachments/assets/78f4893c-8246-4d44-8489-c8e70992ef28) |  ![image](https://github.com/user-attachments/assets/b3aa1ba8-a0f5-4bbb-b8c8-08699d423742) |
| Animation, `BlendSpace2D` (horizontal and vertical overflow, WIP) | ![image](https://github.com/user-attachments/assets/9ac9fdea-66e3-4056-bb21-84dd2f62648e) | ![image](https://github.com/user-attachments/assets/be256ba8-e3ff-4111-a78f-e4a11bfc469c) |
| `Polygon2DEditor` (horizontal and vertical overflow) | ![image](https://github.com/user-attachments/assets/524c114b-acde-477a-a68d-6451e0d97e8b) | ![image](https://github.com/user-attachments/assets/fe813a3e-9a18-4f19-a441-45c8551b8661) |
| `Shader Editor` (vertical overflow) | ![image](https://github.com/user-attachments/assets/0771278f-ac54-42c9-8d89-1d530844dd6f) | ![image](https://github.com/user-attachments/assets/3adb3fa8-a377-4369-a82c-41c60ae8bd75) |
| `ShaderFile` (vertical overflow) | ![image](https://github.com/user-attachments/assets/9a10cbb7-24a9-4bb6-b739-ed6810253c5d) | ![image](https://github.com/user-attachments/assets/b0e86566-e154-4c69-97af-7b7ea613ccc7) |
| `SpriteFrames` (vertical overflow) | ![image](https://github.com/user-attachments/assets/c66898ad-9fae-494b-baae-cffc5e6c2d8e) | ![image](https://github.com/user-attachments/assets/0920b4bb-70c9-4633-8da0-e636c332224b) |
| `ThemeEditor` (horizontal and vertical overflow) | ![image](https://github.com/user-attachments/assets/1c324375-dcc5-4799-b528-ad9fb767e074) | ![image](https://github.com/user-attachments/assets/281629f9-97fe-4d67-b10c-3717c5de6b81) |
| `EditorNetworkProfiler` (vertical overflow) | ![image](https://github.com/user-attachments/assets/3767443e-25a0-44ee-9ae3-30385c552711) | ![image](https://github.com/user-attachments/assets/09729706-d0f2-46f8-80b0-9349e0eaf3ee) |

arkology committed:
Minimize some editor panels

KoBeWi commented on editor/editor_audio_buses.cpp:
```diff
@@ -1337,7 +1337,8 @@ EditorAudioBuses::EditorAudioBuses() {
 	bus_scroll->set_v_size_flags(SIZE_EXPAND_FILL);
-	bus_scroll->set_vertical_scroll_mode(ScrollContainer::SCROLL_MODE_DISABLED);
+	bus_scroll->set_custom_minimum_size(Size2(0, 200) * EDSCALE);
+	bus_scroll->set_size(Size2(0, 350) * EDSCALE);
```
I don't think `set_size()` has any effect here.
![image](https://github.com/user-attachments/assets/873b3947-c783-4678-bdfe-0c9d2c4772b5)
Minimizing the editor panels should not affect editor at full size. We need a better sizing mechanism before doing such changes.

arkology replied:
I think I have an idea about it. I will try it and then write the results here.

arkology replied:
@KoBeWi, I made this https://github.com/godotengine/godot/pull/109915. I think in theory it can be used in all other editors too.

KoBeWi commented:
This makes some editors too small by default (as noted above). Unfortunately we don't have "preferred size" and things usually shrink to minimum size. Trying to set default size won't have effect. I encountered the same problem while trying to resolve #26392 (by making minimum size smaller, with default being the same as now) and there is just no solution right now.

arkology commented:
Thanks for review!
So, at the moment, the only way this pull request can help is to identify some of the places that cause the GUI overflow bug. I think I have an idea about how minimum_size could be changed without breaking current behaviour, bug not sure it will work.

Some findings (for Tilemap/TileSet editors, for example) may be still valid and I will move them to another PR. 
@KoBeWi what do you think, are current changes OK for Polygon2DEditor, TileMapLayerEditorTilesPlugin, TileSetEditor, EditorNetworkProfiler?

arkology commented:
> In Network Profiler the Up label should be together with the following field.

I completely agree!

Calinou reviewed (commented):
Tested locally (rebased on top of `master` b607110ad211778a6426a00ca9c8cd99c0eb383c), it works as expected.

However, there's the issue that the Audio bottom panel now looks like this when you first open it, regardless of the window height:

![image](https://github.com/user-attachments/assets/01e262f0-13b9-4408-a73a-0ff83eef2e6e)

We should try to use a "suggested" height on a best-effort basis where you can see the whole thing, and only see a smaller height if we physically can't fit it within the editor window without overflowing.

I'm not sure if the concept of default/suggested height should be present at a (Split)Container level, or just specific to the editor. Many non-game applications made with Godot run into similar issues in my experience, so they end up having to add some bespoke code for it.

This is the smallest possible height without a visible scroll bar:

![image](https://github.com/user-attachments/assets/8977d4dc-e8bf-4cc3-b650-94f06f4ca865)

(Sometimes, I wonder if we could redesign the Audio bottom panel to use an horizontal layout, but this would come with its own caveats when working with a lot of buses.)

WhalesState commented:
> However, there's the issue that the Audio bottom panel now looks like this when you first open it, regardless of the window height:
> 
> ![image](https://private-user-images.githubusercontent.com/180032/413391170-01e262f0-13b9-4408-a73a-0ff83eef2e6e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQ1MDczMjEsIm5iZiI6MTc1NDUwNzAyMSwicGF0aCI6Ii8xODAwMzIvNDEzMzkxMTcwLTAxZTI2MmYwLTEzYjktNDQwOC1hNzNhLTBmZjgzZWVmMmU2ZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgwNlQxOTAzNDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zMGZhNDExZjdkMGI0ODE3Yjc0Y2QzZDlmZmFmOGJiZjhhMjY2ZGU1OTljOWU1MjE5NWI2ZTdhMzM3OTgxYWI2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.pyhZrpn1kM5uqrDoYsgCw_NEm0BbfzcINe3hd1FAm-4)

How about queue disabling/enabling vertical scroll when the Audio bottom panel becomes visible/hidden ? will try to see if this can  fix the issue.

arkology commented on editor/editor_audio_buses.cpp:
```diff
@@ -1337,7 +1337,8 @@ EditorAudioBuses::EditorAudioBuses() {
 	bus_scroll->set_v_size_flags(SIZE_EXPAND_FILL);
-	bus_scroll->set_vertical_scroll_mode(ScrollContainer::SCROLL_MODE_DISABLED);
+	bus_scroll->set_custom_minimum_size(Size2(0, 200) * EDSCALE);
+	bus_scroll->set_size(Size2(0, 350) * EDSCALE);
```
Now I have a bit more experience in this area and I know this :)

arkology commented:
<img width="572" height="51" alt="image" src="https://github.com/user-attachments/assets/cb1494c1-d39b-4b11-ba98-0051dcc63034" />

Note for Godot team members: No need to review this PR, I split it into individual PRs linked above (not everything is done at the moment, but most of the work has already been done.)
UPD: Converted PR to draft. Will close it once all its parts will be separated into individual PRs and merged.
